cli/tests/test_diff_command.rs

```diff
@@ -740,6 +740,83 @@ fn test_diff_skipped_context_nondefault() {
     "###);
 }
 
+#[test]
+fn test_diff_formatter_setting() {
```

Comment:
While I also adjusted the existing tests to use the new setting key, this new test just verifies that the various builtin and external formats can be used (or an appropriate error is shown).